lib/helper/Testcafe.js

```diff
@@ -0,0 +1,661 @@
+// @ts-nocheck
+const fs = require('fs');
+const assert = require('assert');
+const path = require('path');
+const requireg = require('requireg');
+const createTestCafe = require('testcafe');
+const { Selector, ClientFunction } = require('testcafe');
+
+const testControllerHolder = require('./testcafe/testControllerHolder');
+const { mapError, createTestFile } = require('./testcafe/testcafe-utils');
+
+const stringIncludes = require('../assert/include').includes;
+const { urlEquals } = require('../assert/equal');
+const {
+  xpathLocator,
+} = require('../utils');
+const Locator = require('../locator');
+const Helper = require('../helper');
+
+/**
+ * Client Functions
+ */
+const getPageUrl = t => ClientFunction(() => document.location.href).with({ boundTestRun: t });
+
+/**
+ * Uses Testcafe to run tests.
+ */
+class Testcafe extends Helper {
+  constructor(config) {
+    super(config);
+
+    this.iteration = 1;
+    this.testcafe = undefined; // testcafe instance
+    this.t = undefined; // testcafe test controller
+    this.dummyTestcafeFile; // generated testcafe test file
+
+    this.context = undefined; // TODO Not sure if this applies to testcafe
+
+
+    this.options = Object.assign({
+      url: 'http://localhost',
+      show: false,
+      browser: 'chrome',
+      restart: true, // TODO Test if restart false works
+      manualStart: false,
+      keepBrowserState: false,
+      waitForTimeout: 10000,
+      fullPageScreenshots: false,
+      disableScreenshots: false,
+      windowSize: undefined,
+    }, config);
+  }
+
+  // TOOD Do a requirements check
+  static _checkRequirements() {
+    try {
+      requireg('testcafe');
+    } catch (e) {
+      return ['testcafe@^1.1.0'];
+    }
+  }
+
+  async _startBrowser() {
+    this.dummyTestcafeFile = createTestFile(); // create a dummy test file to get hold of the test controller
+
+    this.iteration += 2; // Use different ports for each test run
+    // @ts-ignore
+    this.testcafe = await createTestCafe('localhost', 1338 + this.iteration, 1339 + this.iteration);
+
+    this.debugSection('_before', 'Starting testcafe browser...');
+
+    this.isRunning = true;
+
+    // TODO Do we have to cleanup the runner
+    const runner = this.testcafe.createRunner();
+    runner
+      .src(this.dummyTestcafeFile)
+      .screenshots(global.output_dir, true)
+      // .video(global.output_dir)
```

Comment:
I think video should be enabled in config

Reply:
Makes sense